Fatras/include/ActsFatras/Kernel/Simulation.hpp

```diff
@@ -49,13 +49,22 @@ struct SingleParticleSimulation {
   interactions_t interactions;
   /// Selector for surfaces that should generate hits.
   hit_surface_selector_t selectHitSurface;
+  /// Wrapped logger for debug output.
+  Acts::LoggerWrapper Logger = Acts::getDummyLogger();
   /// Local logger for debug output.
   std::shared_ptr<const Acts::Logger> localLogger = nullptr;
 
   /// Construct the simulator with the underlying propagator.
   SingleParticleSimulation(propagator_t &&propagator_, Acts::Logging::Level lvl)
       : propagator(propagator_),
-        localLogger(Acts::getDefaultLogger("Simulation", lvl)) {}
+        localLogger(Acts::getDefaultLogger("Simulation", lvl)) {
+    Logger = Acts::LoggerWrapper(*localLogger);
+  }
+
+  /// Alternatively construct the simulator with an external logger.
+  SingleParticleSimulation(propagator_t &&propagator_,
+                           Acts::LoggerWrapper Logger_)
+      : propagator(propagator_), Logger(Logger_) {}
 
   /// Provide access to the local logger instance, e.g. for logging macros.
   const Acts::Logger &logger() const { return *localLogger; }
```

Comment:
Wouldn't you want to return the logger inside the `loggerWrapper` to have one consistent output logging facility?

Reply:
My original idea was to not have a localLogger member in Simulation.hpp, but just pass LoggerWrappers. The localLogger was kept to be compatible with old tests. But if you feel like this localLogger should be kept, I will return it here.

Reply:
In principle it would be more consistent that either `SingleParticleSimulation` **does** OR **doesn't** own its logger. So I would either stick to `localLogger`, or accept a `LoggerWrapper` in the constructor and no alternative,  in which case it never owns the logger at all.

Reply:
Since we pass in a `LoggerWrapper` from Athena, I suggest we only have `LoggerWrapper` and drop the `localLogger`

Reply:
If we use keep only `LoggerWrapper` , do we still need two constructors? When using acts-fatras from another software (i.e. Athena), I can foresee that mostly only `LoggerWrapper` will be passed to the constructor. If we agree that we don't need a constructor that accepts `Logger` or `Logging::Level`, I (or someone else) can go ahead and rewrite the unit tests accordingly.

Reply:
`LoggerWrapper` never owns it's logger. In other components of ACTS in Athena, a `std::unique_ptr<Logger>` is created using an Athena specific adaptor, and passed to the component. I think this would work for Fatras two. In that case, we could simplify to **one** constructor, which accepts a `std::unique_ptr<Logger>` (or possibly a `std::shared_ptr<Logger>` if really necessary), and create `LoggerWrapper` on the fly to pass to the propagator.

Reply:
ok, I agree with this implementation as well.

Reply:
@paulgessinger @asalzburger 
I changed the constructor to take only take std::unique_ptr<Logger>, please review the new changes and give me some feedback

Reply:
Hey @yifever, from what I can tell it now accepted `std::shared_ptr<Logger>`. Is that intentional?

Reply:
I followed the current implementation in master. I just thought that since the local logger member is a `shared_ptr` (this is true right now before merge) then it should accept a `shared_ptr`. This can be easily changed if you think it should be.

Reply:
You're right. Consider using `std::move` to avoid copying the shared pointer another time. I'll resolve this discussion.